x-pack/plugins/security_solution/server/lib/telemetry/artifact.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import axios from 'axios';
+import AdmZip from 'adm-zip';
+import type { ITelemetryReceiver } from './receiver';
+import type { ESClusterInfo } from './types';
+
+export interface IArtifact {
+  start(receiver: ITelemetryReceiver): Promise<void>;
+  getArtifact(name: string): Promise<unknown>;
+}
+
+class Artifact implements IArtifact {
+  private manifestUrl?: string;
+  private readonly CDN_URL = 'https://artifacts.security.elastic.co';
+  private readonly AXIOS_TIMEOUT_MS = 10_000;
+  private receiver?: ITelemetryReceiver;
+  private esClusterInfo?: ESClusterInfo;
+
+  public async start(receiver: ITelemetryReceiver) {
+    this.receiver = receiver;
+    this.esClusterInfo = await this.receiver.fetchClusterInfo();
+    const version = this.esClusterInfo?.version?.number;
+    this.manifestUrl = `${this.CDN_URL}/downloads/endpoint/manifest/artifacts-${version}.zip`;
+  }
+
+  public async getArtifact(name: string): Promise<unknown> {
+    if (this.manifestUrl) {
+      const response = await axios.get(this.manifestUrl, {
+        timeout: this.AXIOS_TIMEOUT_MS,
+        responseType: 'arraybuffer',
+      });
+      const zip = new AdmZip(response.data);
+      const entries = zip.getEntries();
+      const manifest = JSON.parse(entries[0].getData().toString());
+      const relativeUrl = manifest.artifacts[name].relative_url;
+      if (relativeUrl) {
+        const url = `${this.CDN_URL}${relativeUrl}`;
+        const artifactResponse = await axios.get(url, { timeout: this.AXIOS_TIMEOUT_MS });
+        return artifactResponse.data;
+      } else {
+        throw Error(`No artifact for name ${name}`);
```

Comment:
When this error is thrown, it is just caught by the task manager right? How loud is this in the logs?

Reply:
It's caught by the task and logged by tlog. In terms of noise it would log every 30 mins(if failing everytime). Great point about falling back to the default if any errors in applying occur!

Reply:
Are you sure this would be logged by `tlog`, and not `log`? 
I think it will end up in users' logs regardless because we are throwing an error. I also think it would be worthwhile being a little more descriptive.